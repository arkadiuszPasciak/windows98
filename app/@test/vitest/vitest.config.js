import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"
import tsconfigPaths from "vite-tsconfig-paths"

export default mergeConfig(
	sharedConfig,
	defineProject({
		plugins: [tsconfigPaths({ loose: true })],
	})
)
