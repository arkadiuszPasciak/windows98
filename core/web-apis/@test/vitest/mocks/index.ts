export * from "./canvas-api.mock"
export * from "./media-devices-api.mock"
export * from "./screen-api.mock"
export * from "./video-api.mock"
