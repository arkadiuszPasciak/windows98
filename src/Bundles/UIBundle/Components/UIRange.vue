<template>
  <div class="UIRange" :class="`variant-${variant} position-${position}`">
    <label v-if="labelNameStart" class="label is-start" :for="id">{{
      labelNameStart
    }}</label>

    <label
      v-if="labelNameMiddle && position === 'vertical'"
      class="label is-middle"
      :for="id"
      >{{ labelNameMiddle }}</label
    >

    <input
      :id="id"
      class="range"
      type="range"
      :min="minValue"
      :max="maxValue"
      :step="stepValue"
      :value="modelValue"
      @change="
        $emit('update:modelValue', ($event.target as HTMLInputElement).value)
      "
    />

    <label v-if="labelNameEnd" class="label is-end" :for="id">{{
      labelNameEnd
    }}</label>
  </div>
</template>

<script setup lang="ts">
  import { PropType } from 'vue'

  defineProps({
    variant: {
      type: String as PropType<'primary'>,
      default: 'primary',
    },
    id: {
      type: String,
      required: true,
    },
    minValue: {
      type: Number,
      default: 0,
    },
    maxValue: {
      type: Number,
      default: 10,
    },
    stepValue: {
      type: Number,
      default: 1,
    },
    modelValue: {
      type: Number,
      required: true,
    },
    position: {
      type: String as PropType<'horizontal' | 'vertical'>,
      default: 'horizontal',
    },
    labelNameStart: {
      type: String,
      required: false,
    },
    labelNameMiddle: {
      type: String,
      required: false,
    },
    labelNameEnd: {
      type: String,
      required: false,
    },
  })

  defineEmits(['update:modelValue'])
</script>

<style lang="scss" scoped>
  @import '@Bundles/UIBundle/Styles/UIRange.styles.scss';
</style>
