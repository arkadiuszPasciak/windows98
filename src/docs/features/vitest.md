# [Vitest](https://vitest.dev/)

## Description

Vitest is a modern and fast testing framework specifically designed for Vite-powered projects. Developed by the [Vite team](https://vitejs.dev/team.html), Vitest leverages Vite's architecture to provide a seamless testing experience with features like native ESM support, rapid test execution, and isolated testing environments. It offers compatibility with various front-end libraries and frameworks, such as Vue, React, and Svelte. With its straightforward configuration and integration with Vite, Vitest aims to enhance the developer experience by providing an efficient and enjoyable testing workflow for Vite-based web applications.

## Commands

[Available commands in the application](/reference/cli.html#vitest)
