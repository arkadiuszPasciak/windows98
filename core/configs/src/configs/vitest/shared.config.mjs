import { defineConfig } from 'vitest/config'

export default defineConfig({
    test: {
        environment: "jsdom",
        exclude: [
            "**/build/**",
            "**/dist/**",
            "**/node_modules/**",
            "**/@bundler/**",
            "**/@linter/**",
            "**/@test/**",
        ],
        include: [
            "**/*.unit.ts"
        ],
        globals: true,
    },
})
