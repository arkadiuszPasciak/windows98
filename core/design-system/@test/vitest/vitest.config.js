import { defineProject, mergeConfig } from "vitest/config"
import sharedConfig from "@windows98/configs/vitest/shared"

export default mergeConfig(
	sharedConfig,
	defineProject({/* here custom configs */ })
)
