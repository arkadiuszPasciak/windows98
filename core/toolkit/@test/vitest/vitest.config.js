import sharedConfig from "@windows98/configs/vitest/shared"
import { defineProject, mergeConfig } from "vitest/config"

export default mergeConfig(
	sharedConfig,
	defineProject({
		/* here custom configs */
	}),
)
