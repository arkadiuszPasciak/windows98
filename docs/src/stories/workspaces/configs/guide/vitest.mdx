import { Meta, Title } from "@storybook/addon-docs/blocks";

<Meta title="Workspaces/Configs/Guide/Vitest" />

<Title>âš¡ Vitest</Title>

Complete guide to using Vitest configuration from `@windows98/configs` for unit and integration testing.

## Overview

Vitest configuration provides fast, modern testing setup optimized for TypeScript and modern JavaScript frameworks within the Windows98 ecosystem.

## Configuration Path

```
@windows98/configs/vitest/shared
```

### Install Dependencies

```bash
# Install vitest
pnpm add -D vitest

# Install the configs package
pnpm add -D @windows98/configs
```

## Basic Usage

### Simple Setup

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig(sharedConfig)
```

### With Custom Settings

```javascript
// ./@test/vitest/vitest.config.js
import { defineConfig } from 'vitest/config'
import sharedConfig from '@windows98/configs/vitest/shared'

export default defineConfig({
  ...sharedConfig,
  test: {
    ...sharedConfig.test,
    environment: 'jsdom',
    setupFiles: ['./src/test-setup.ts']
  }
})
```

## Package.json Scripts

```json
{
  "scripts": {
	"test:unit:run": "vitest run -c ./@test/vitest/vitest.config.js",
	"test:unit:watch": "vitest -c ./@test/vitest/vitest.config.js",
  }
}
```

