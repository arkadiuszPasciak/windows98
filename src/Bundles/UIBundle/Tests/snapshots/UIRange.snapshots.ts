import { mount } from '@vue/test-utils'
import { expect, it, describe } from 'vitest'
import UIRange from '@Bundles/UIBundle/Components/UIRange.vue'

const horizontalPrimaryWrapper = mount(UIRange, {
  props: {
    variant: 'primary',
    id: 'person-age',
    modelValue: 1,
    position: 'horizontal',
    labelNameStart: 'Low',
    labelNameEnd: 'High',
    minValue: 1,
    maxValue: 12,
    stepValue: 3,
  },
})

const verticalPrimaryWrapper = mount(UIRange, {
  props: {
    variant: 'primary',
    id: 'something-number',
    modelValue: 5,
    position: 'vertical',
    labelNameStart: 'Start',
    labelNameEnd: 'End',
    minValue: 5,
    maxValue: 20,
    stepValue: 1,
  },
})

describe('[UIBundle]<Snapshots>(UIRange)', async () => {
  it('renders correctly component: horizontal with primary', () => {
    expect(horizontalPrimaryWrapper.element).toMatchSnapshot()
  })

  it('renders correctly component: vertical with primary', () => {
    expect(verticalPrimaryWrapper.element).toMatchSnapshot()
  })
})
