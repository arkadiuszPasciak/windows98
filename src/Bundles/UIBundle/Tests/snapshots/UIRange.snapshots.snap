// Vitest Snapshot v1

exports[`[UIBundle]<Snapshots>(UIRange) > renders correctly component: horizontal with primary 1`] = `
<div
  class="UIRange variant-primary position-horizontal"
  data-v-085e19e1=""
>
  <label
    class="label is-start"
    data-v-085e19e1=""
    for="person-age"
  >
    Low
  </label>
  <!--v-if-->
  <input
    class="range"
    data-v-085e19e1=""
    id="person-age"
    max="12"
    min="1"
    step="3"
    type="range"
  />
  <label
    class="label is-end"
    data-v-085e19e1=""
    for="person-age"
  >
    High
  </label>
</div>
`;

exports[`[UIBundle]<Snapshots>(UIRange) > renders correctly component: vertical with primary 1`] = `
<div
  class="UIRange variant-primary position-vertical"
  data-v-085e19e1=""
>
  <label
    class="label is-start"
    data-v-085e19e1=""
    for="something-number"
  >
    Start
  </label>
  <!--v-if-->
  <input
    class="range"
    data-v-085e19e1=""
    id="something-number"
    max="20"
    min="5"
    step="1"
    type="range"
  />
  <label
    class="label is-end"
    data-v-085e19e1=""
    for="something-number"
  >
    End
  </label>
</div>
`;
